### PR TITLE
Run html/css validation on src files

### PR DESCRIPTION
**Pull request by mvr320**
This scrapes the testinstance to plain html and checks the usage of HTML not in line with W3C recommendations. Currently it would only break when more errors are found than currently in the main branch.

Improvement is possible in a number of ways;
- Testing with a user which has a team role, As I think the admin user does not have this.
- Some code is duplicated from the integration tests, it would help if these are moved to another file which can than be sourced, I'm fine with trying something like that but would think that should be in another PR

The output is now divided into sections both for the file and its problems and another for a problem and which files this occurs as I think fixing /public should be the 1st one but fixing 1 problem at a time might be safer to not get differences in the UI.

All these problems on itself can be labeled as "Good first time" issues in my opinion.

If this is accepted I'll fixup to only 1 commit.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -26,6 +26,35 @@ check_w3c_css:
...
+    MYSQL_ROOT_PASSWORD: password
+    DOCKER_HOST: tcp://docker:2375/
+    DOCKER_DRIVER: overlay2
+  script:
+  - |
+    docker run --privileged \
+      -v $CI_PROJECT_DIR:$CI_PROJECT_DIR \
+      --net=host \
+      -e "TERM=xterm-256color" -e "HOME=$HOME" -e "USER=$USER" -e "MARIADB_PORT_3306_TCP_ADDR=$MARIADB_PORT_3306_TCP_ADDR" -e "MYSQL_ROOT_PASSWORD=$MYSQL_ROOT_PASSWORD" \
+      mvr320/djcihtml:latest /bin/bash -eo pipefail -c "umask 0002; cd $CI_PROJECT_DIR; script --return -qfc \"./gitlab/html.sh\" /dev/null | ts \"[%F %T]\" "
```
Comment (nickygerritsen) [inline .gitlab-ci.yml:50]: Probably better to push this under the domjudge user. Also we can probably keep the source in domjudge-script?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -26,6 +26,35 @@ check_w3c_css:
...
+  artifacts:
+    when: always
+    paths:
```
Comment (nickygerritsen) [inline .gitlab-ci.yml:53]: Double paths

**Review by nickygerritsen — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (mvr320) [reply .gitlab-ci.yml:50]: Yes when the output was accepted I was going to start making PRs to the gitlabci project assuming thats the source you wanted there

Comment (mvr320) [reply .gitlab-ci.yml]: This one:
https://github.com/DOMjudge/domjudge-packaging/pull/63

**Review by nickygerritsen — APPROVED** [review]

**Pull request merged**